pytorch_lightning/trainer/optimizers.py

```diff
@@ -42,7 +42,12 @@ def init_optimizers(self, model: LightningModule) -> Tuple[List, List, List]:
         if isinstance(optim_conf, Optimizer):
             optimizers = [optim_conf]
         # two lists, optimizer + lr schedulers
-        elif isinstance(optim_conf, (list, tuple)) and len(optim_conf) == 2 and isinstance(optim_conf[0], list):
+        elif (
+            isinstance(optim_conf, (list, tuple))
+            and len(optim_conf) == 2
+            and isinstance(optim_conf[0], list)
+            and all(isinstance(opt, Optimizer) for opt in optim_conf[0])
```

Comment:
Any reason in particular why you added the `isinstance` check?

Reply:
Yes this is to prevent configurations like
```python
[{'optimizer': optimizer1}, {'optimizer': optimizer2}], [lr_scheduler1, lr_scheduler2]
```

which will cause errors later down the line.